lesson16/js/main.js

```diff
@@ -0,0 +1,188 @@
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function createTabContent(){
+	const newsContentArea = document.createElement('div');
+	newsContentArea.classList.add('news-area');
+	newsContentArea.setAttribute('role','tabpanel');
+	categoryTab.parentNode.insertBefore(newsContentArea,categoryTab.nextSibling);
+}
+
+function renderCategoryTab(data){
+	const fragment = document.createDocumentFragment();
+
+	data.forEach((category) => {
+		const categoryTab = document.createElement('li');
+		categoryTab.classList.add('tab_list_item');
+		categoryTab.setAttribute('role','presentation');
+
+		const categoryButton = document.createElement('button');
+		categoryButton.textContent = category.name;
+		categoryButton.setAttribute('class', 'tab-button js-tabButton');
+		categoryButton.setAttribute('aria-selected', 'false');
+		categoryButton.setAttribute('role', 'tab');
+		categoryButton.setAttribute('tabindex', '-1');
+		fragment.appendChild(categoryTab).appendChild(categoryButton);
+	});
+
+	const categoryTabLists = document.querySelector('.tab-list');
+	categoryTabLists.appendChild(fragment);
+}
+
+function categoryTabInitial(){
+	const firstTab = document.querySelectorAll('.tab-button')[0];
+	firstTab.setAttribute('aria-selected','true');
+	firstTab.setAttribute('tabindex','0');
+}
+
+function renderNewsArticle(data){
+	const newsContentArea = document.querySelector('.news-area');
+	const newsLists = document.createElement('ul');
+	newsLists.classList.add('news-list');
+	const fragment = document.createDocumentFragment();
+
+	const displayedResults = 5;
+	for (let i = 0; i < displayedResults; ++i){
+		const li = document.createElement('li');
+		li.classList.add('news-item');
+
+		const anchor = document.createElement('a');
+		anchor.classList.add('news-link');
+		anchor.href = data[i].link;
+		anchor.textContent = data[i].title;
+		fragment.appendChild(li).appendChild(anchor);
+	}
+
+	newsContentArea.appendChild(newsLists).appendChild(fragment);
+};
+
+function resetArticle(){
+	const newsLists= document.querySelector('.news-list');
+	newsLists.remove();
+}
+
+function displayErrorMassage(error){
+	const getNewsArea = document.querySelectorAll('.news-area');
+	for(let i = 0; i < getNewsArea.length; i++){
+		const createTextBox = document.createElement('p');
+		createTextBox.textContent = error;
+		getNewsArea[i].appendChild(createTextBox);
+	}
+}
+
+async function fetchNewsData() {
+	const NEWS_DATA_URL = 'https://api.json-generator.com/templates/2PqhEvPcqUZW/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl';
+	try {
+		const response = await fetch(NEWS_DATA_URL);
+		if(!response.ok){
+			throw new Error('Network Error');
+		} else{
+			const json = await response.json();
+			return json;
+		}
+	}
+	catch (error){
+		console.error('Failed to get the data. error:',error);
+		displayErrorMassage(error);
+	}
+}
```

Comment:
MUST
After moving the tab a few times, I get an error. Please investigate the cause.
<img width="749" alt="スクリーンショット 2022-04-05 14 26 44" src="https://user-images.githubusercontent.com/43788628/161684804-3119f9e8-d816-45c1-af74-8b1fbb69d3de.png">

Reply:
Thank you for your confirmation!
When I checked the error, it was a status 429 error.
I forgot to write the argument in try-catch, so I fixed it only here.
![スクリーンショット 0004-04-06 4 45 46](https://user-images.githubusercontent.com/75970215/161838655-6dcb7634-f006-4325-a528-ee79822047b3.png)

[c4739e4](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/c4739e45bf7d8b98ff4070e30297d415abcb26a5)

Since I got a 429 error, I will correct the load for each click you pointed out to the first time.

Reply:
Changed to read data only once.
We just took the method of storing the retrieved data in a global variable, but we were not sure if this was the best way to do it.

Please check this and other information.

[48b4fda](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/48b4fdad9b5fe3a71aa6d76bdbf99f3b9629a604)